src/main/java/org/killbill/billing/plugin/adyen/client/model/paymentinfo/KlarnaPaymentInfo.java

```diff
@@ -51,13 +54,52 @@ private KlarnaPaymentInfo(KlarnaPaymentInfoBuilder builder) {
         this.setPaymentType(builder.paymentType);
     }
 
+    public boolean completeKlarnaAuthorisation() {
+        return detailsData.size() > 0;
+    }
+
+    private void setAuthResultKeys(Map<String, String> additionalData) {
+        List<String> authCompleteKeys = new ArrayList<>();
+        String authResultKeys = additionalData.get("resultKeys");
+        if(authResultKeys != null) {
+            try {
+                Map<String, String> authKeyMap = mapper.readValue(authResultKeys, Map.class);
+                authCompleteKeys.addAll(authKeyMap.keySet());
+            } catch (IOException ex) {
+                logger.warn("Failed to retrieve details keys: " + ex.getMessage());
```

Comment:
This is very specific to Klarna, can we add that detail in the log to avoid confusion.